jme3-core/src/main/java/com/jme3/texture/TextureImage.java

```diff
@@ -0,0 +1,308 @@
+         * 
+         * @return 
+         */
+        public int getGlEnum() {
+            return glEnum;
+        }
+        
+    }
+    
+    private Texture texture;
+    private int level, layer;
+    private Access access;
+    private boolean updateFlag = true;
+    
+    public TextureImage(Texture texture) {
+        this(texture, 0, -1, Access.ReadWrite);
+    }
+    public TextureImage(Texture texture, Access access) {
+        this(texture, 0, -1, access);
+    }
+    public TextureImage(Texture texture, int level, int layer) {
+        this(texture, level, layer, Access.ReadWrite);
+    }
+    public TextureImage(Texture texture, int level, int layer, Access access) {
+        this.texture = Objects.requireNonNull(texture, "Underlying texture cannot be null");
+        this.level = level;
+        this.layer = layer;
+        this.access = access;
+        if (this.level < 0) {
+            throw new IllegalArgumentException("Level cannot be less than zero.");
+        }
+    }
+    
+    /**
+     * Binds this texture image to the texture unit.
+     * <p>
+     * Calling this is not completely sufficient for totally binding an image
+     * to an image unit. Additionally, the image must be bound beforehand using
+     * {@link GL2#glBindTexture(int, int)}.
+     * 
+     * @param gl4 OpenGL4 implementation (not null)
+     * @param texUtil utility used to convert JME's image format to the corresponding GL enum (not null)
+     * @param unit texture unit to bind to
+     */
+    public void bindImage(GL4 gl4, TextureUtil texUtil, int unit) {
+        Image img = texture.getImage();
+        gl4.glBindImageTexture(unit, img.getId(), level, isLayered(), Math.max(layer, 0),
+                access.getGlEnum(), texUtil.getImageFormat(img.getFormat(), false).internalFormat);
+    }
+    
+    /**
+     * Sets the update flag indicating this texture image needs rebinding.
+     */
+    public void setUpdateNeeded() {
+        updateFlag = true;
+    }
+    
+    /**
+     * Clears the update flag and returns the update flag's value before
+     * it was cleared.
+     * 
+     * @return 
+     */
+    public boolean clearUpdateNeeded() {
+        boolean f = updateFlag;
+        updateFlag = false;
+        return f;
+    }
+    
+    /**
+     * Sets the underlying texture wrapped by this TextureImage.
+     * 
+     * @param texture 
+     */
+    public void setTexture(Texture texture) {
+        Objects.requireNonNull(texture, "Wrapped texture cannot be null.");
+        if (this.texture != texture) {
+            this.texture = texture;
+            updateFlag = true;
+        }
+    }
+    
+    /**
+     * Sets the image level to bind.
+     * <p>
+     * The level controls which mipmap level will be bound to the texture unit,
+     * where zero corresponds to the base level of the texture.
+     * 
+     * @param level level to bind (cannot be negative)
+     */
+    public void setLevel(int level) {
```

Comment:
I would suggest a more explicit Javadoc to clarify the method's functionality:

e.g.
```java
/**
 * Sets the mipmap level to bind to the texture unit.
 * <p>
 * The level parameter specifies which mipmap level will be bound, where zero
 * corresponds to the base level of the texture.
 * </p>
 * 
 * @param level the mipmap level to bind; must not be negative
 * @throws IllegalArgumentException if the specified level is negative
 */
```